src/structarray.jl

```diff
@@ -445,7 +445,11 @@ end
 # broadcast
 import Base.Broadcast: BroadcastStyle, ArrayStyle, AbstractArrayStyle, Broadcasted, DefaultArrayStyle
 
-struct StructArrayStyle{Style} <: AbstractArrayStyle{Any} end
+struct StructArrayStyle{S,N} <: AbstractArrayStyle{N} end
+# If `S` also track input's dimensionality, we'd better also update it.
+StructArrayStyle{S,M}(::Val{N}) where {M,S<:AbstractArrayStyle{M},N} =
+    StructArrayStyle{typeof(S(Val(N))),N}()
```

Comment:
Could this maybe be written using the `function StructArrayStyle(...)` syntax and be split over a couple of lines, for the sake of readability?